### PR TITLE
Fix or suppress some compilation warnings

### PR DESCRIPTION
**Pull request by dwijnand**
Slowly starting to select, cleanup and push all my local warning fixes and cleanups.  This is part 1 of ??.

### CODE REVIEW

**Comment by dwijnand** [comment]
Happy to answer to feedback and happy to amend the PR, including deferring some of the changes to a later date (later PR).  Speedy reviews desired.

**Comment by dwijnand** [comment]
```
[info] Compiling 176 Scala sources and 136 Java sources to /home/travis/build/playframework/playframework/core/play/target/scala-2.13.0-M5/classes ...
[error] /home/travis/build/playframework/playframework/core/play/src/main/scala/play/api/data/Form.scala:105:26: type mismatch;
[error]  found   : Iterable[(String, Seq[String])]
[error]  required: Map[String,Seq[String]]
[error]     bindFromRequest(data ++ method)
[error]                          ^
[error] one error found
```

File: core/play/src/test/scala/play/api/libs/TemporaryFileCreatorSpec.scala [hunk 1]
```diff
@@ -406,7 +406,7 @@ class TemporaryFileCreatorSpec extends Specification with Mockito {
         writeFile(file, "file to be moved")
 
         val destination = parentDirectory.resolve("destination.txt")
-        creator.create(file).atomicMoveWithFallback(destination)
+        creator.create(file).atomicMoveFileWithFallback(destination)
```
Comment (mkurz) [inline core/play/src/test/scala/play/api/libs/TemporaryFileCreatorSpec.scala]: Should we rename those methods back to their original name in 2.8? Deprecating `atomicMoveFileWithFallback` and `moveFileTo`, naming them `atomicMoveWithFallback` and `moveTo` again?

File: core/play/src/test/java/play/mvc/RangeResultsTest.java [hunk 2]
```diff
@@ -361,20 +362,22 @@ public void shouldReturnRangeResultForStreamWhenFilenameHasSpecialChars() throws
...
 
     mockRegularFileTypes();
+    return request;
   }
 
-  private void mockRangeRequest() {
+  private Http.Request mockRangeRequest() {
     Http.Request request = mock(Http.Request.class);
     when(request.header(RANGE)).thenReturn(Optional.of("bytes=0-1"));
     when(this.ctx.request()).thenReturn(request);
```
Comment (mkurz) [inline core/play/src/test/java/play/mvc/RangeResultsTest.java]: @dwijnand Now that you pass the request explicitly, we do not need to set the ctx threadlocal anymore, see #9255

File: core/play/src/test/scala/play/api/libs/TemporaryFileCreatorSpec.scala (hunk 1 above)
Comment (mkurz) [reply core/play/src/test/scala/play/api/libs/TemporaryFileCreatorSpec.scala]: Ahh, I already added this to my TODO list :wink: https://github.com/playframework/playframework/issues/8344#issuecomment-455139447

**Pull request merged**